bundles/EcommerceFrameworkBundle/Command/IndexService/ProcessPreparationQueueCommand.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Bundle\EcommerceFrameworkBundle\Command\IndexService;
+
+use Pimcore\Bundle\EcommerceFrameworkBundle\Factory;
+use Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\IndexService;
+use Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\IndexUpdateService;
+use Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\AbstractBatchProcessingWorker;
+use Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\BatchProcessingWorkerInterface;
+use Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\ProductCentricBatchProcessingWorker;
+use Pimcore\Bundle\EcommerceFrameworkBundle\Model\IndexableInterface;
+use Pimcore\Console\Traits\Parallelization;
+use Pimcore\Console\Traits\Timeout;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ProcessPreparationQueueCommand extends AbstractIndexServiceCommand
+{
+    use Timeout;
+    use Parallelization
+    {
+        Parallelization::runBeforeFirstCommand as parentRunBeforeFirstCommand;
+        Parallelization::runAfterBatch as parentRunAfterBatch;
+    }
+
+    /**
+     * @var IndexUpdateService
+     */
+    protected $indexUpdateService;
```

Comment:
just out of curiosity: why is this one protected. The command shouldn't be overriden.

Reply:
maybe not overwritten, but extended?